### PR TITLE
Add Feature Flags capability to Static pages

### PR DESCRIPTION
**Pull request by jesusbotella**
Until now, we didn't have the possibility of loading different scripts/stylesheets depending on a user's feature flag. Due to new dashboard implementation, we need to allow this behaviour. So here it is.

The static script will load assets matching endpoint's feature flags against the ones defined in the user profile.

The feature flag endpoint specification is:

```js
{endpoint}: {
    page: '{endpoint}',
    stylesheets: [
      // Stylesheets
    ],
    scripts: [
      // Scripts
    ],
    feature_flags: {
      '{feature_flag_name}': {
        stylesheets: [
          // Feature flag stylesheets
        ],
        scripts: [
          // Feature flag scripts
        ]
      }
    }
},
```

Related to https://github.com/CartoDB/cartodb/issues/14311

### CODE REVIEW

**Comment by ivanmalagon** [comment]
I guess that whenever you say `endpoint` in your description you mean `entry point`, right?

File: lib/assets/javascripts/dashboard/statics/static.js [hunk 1]
```diff
@@ -28,7 +28,35 @@ window.CartoConfig = window.CartoConfig || {};
     spinnerEl.parentNode.classList.add('is-hidden');
   };
 
-  var addAssets = function () {
+  var getEndpointFeatureFlags = function (staticConfig) {
```
Comment (ivanmalagon) [inline lib/assets/javascripts/dashboard/statics/static.js]:
```suggestion
  var getEntryPointFeatureFlags = function (staticConfig) {
```

File: lib/assets/javascripts/dashboard/statics/static.js [hunk 2]
```diff
@@ -28,7 +28,35 @@ window.CartoConfig = window.CartoConfig || {};
...
+    var featureFlagsData = window.StaticConfig.feature_flags;
```
Comment (ivanmalagon) [inline lib/assets/javascripts/dashboard/statics/static.js]: We should use the argument `staticConfig`, right?

File: lib/assets/javascripts/dashboard/statics/static.js [hunk 3]
```diff
@@ -28,7 +28,35 @@ window.CartoConfig = window.CartoConfig || {};
...
+
+    if (!featureFlagsData) {
+      return [];
+    }
+
+    return Object.keys(featureFlagsData);
+  };
+
+  var getAssetsToLoad = function (userData) {
+    var endpointFeatureFlags = getEndpointFeatureFlags(window.StaticConfig);
```
Comment (ivanmalagon) [inline lib/assets/javascripts/dashboard/statics/static.js]:
```suggestion
    var entryPointFeatureFlags = getEntryPointFeatureFlags(window.StaticConfig);
```

File: lib/assets/javascripts/dashboard/statics/static.js [hunk 4]
```diff
@@ -28,7 +28,35 @@ window.CartoConfig = window.CartoConfig || {};
...
+    var userFeatureFlags = userData.feature_flags;
+
+    var featureFlagToUse = endpointFeatureFlags.filter(function (featureFlag) {
```
Comment (ivanmalagon) [inline lib/assets/javascripts/dashboard/statics/static.js]:
```suggestion
    var featureFlagToUse = entryPointFeatureFlags.filter(function (featureFlag) {
```

**Comment by ivanmalagon** [comment]
Apart from the CR, it looks good. When you make the changes, ping me and I'll do some acceptance in local.

**Comment by jesusbotella** [comment]
Yes, I meant `entry point` 🤦‍♂️. I mixed up the terms 😂

**Comment by ivanmalagon** [comment]
`entry` and `end` are on the edges of the spectrum. It's normal you mistook them. :P

**Pull request merged**